### PR TITLE
ci: setup GitHub Actions

### PR DESCRIPTION
**Pull request by Kikobeats**

### CODE REVIEW

**Comment by kenchris** [comment]
@justinfagnani can you help review this or indicate who might be able to?

File: test/index.js [hunk 1]
```diff
@@ -88,7 +90,9 @@ test('JavaScript URL routing 2/2', t => {
 });
 */
 
-let rawdata = readFileSync('urlpatterntestdata.json');
+
+
```
Comment (kenchris) [inline test/index.js:94]: Why two newlines?

Comment (Kikobeats) [reply test/index.js:94]: this depends of https://github.com/kenchris/urlpattern-polyfill/pull/69, so it should be merged first!

File: package.json [hunk 2]
```diff
@@ -6,28 +6,26 @@
...
     "dist"
   ],
   "devDependencies": {
-    "ava": "^3.14.0",
-    "esm": "^3.2.25",
-    "microbundle": "^0.13.3",
-    "rimraf": "^3.0.2",
-    "rollup": "^2.35.1"
+    "ava": "3",
+    "esm": "latest",
```
Comment (justinfagnani) [inline package.json:19]: These aren't good constraints to use in a checked-in package.json. I'd use lower and upper bounds like they were before.

File: package.json [hunk 3]
```diff
@@ -6,28 +6,26 @@
...
+    "tsup": "latest",
+    "typescript": "latest"
   },
   "scripts": {
-    "build": "rimraf dist/ && microbundle",
-    "build:watch": "microbundle watch",
-    "test": "npm run build && ava --fail-fast -s",
-    "quicktest": "ava --fail-fast -s",
+    "prebuild": "rimraf dist",
+    "build": "tsup src/index.ts --format esm,cjs --dts",
```
Comment (justinfagnani) [inline package.json:26]: This is combining setting up GitHub actions with changing the bundler. Is there a reason to do those together?

Comment (Kikobeats) [reply package.json:26]: these changes are actually part of  https://github.com/kenchris/urlpattern-polyfill/pull/69

Comment (kenchris) [reply package.json:26]: But are you fine with these changes @justinfagnani ?

Comment (justinfagnani) [reply package.json:26]: I don't know tsup, so 🤷‍♂️ :)

**Comment by kenchris** [comment]
We have something similar now, so closing this

**Pull request closed without merging**